proto/message_contents/public_key.proto

```diff
@@ -7,6 +7,34 @@ import "message_contents/signature.proto";
 
 option go_package = "github.com/xmtp/proto/go/message_contents";
 
+// UnsignedPublicKey represents a generalized public key,
+// defined as a union to support cryptographic algorithm agility.
+message UnsignedPublicKey {
+    // The key bytes
+    message Secp256k1Uncompressed {
+        // uncompressed point with prefix (0x04) [ P || X || Y ], 65 bytes
+        bytes bytes = 1; 
+    }
+    uint64 created_ns = 1;
+    oneof union {
+        Secp256k1Uncompressed secp256k1_uncompressed = 2;
+    }
+}
+
+// SignedPublicKey 
+message SignedPublicKey {
+    bytes key_bytes = 1;  // embeds an UnsignedPublicKey
+    Signature signature = 2; // signs key_bytes
+}
+
+// PublicKeyBundle packages the cryptographic keys associated with a wallet,
+// both senders and recipients are identified by their key bundles.
+message SignedPublicKeyBundle {
+    SignedPublicKey identity_key = 1;
+    SignedPublicKey pre_key = 2;
```

Comment:
The `IdentityKey` is signed with the wallet. Is the `pre_key` expected to be signed with the `IdentityKey`? If so, we should note that with a comment.